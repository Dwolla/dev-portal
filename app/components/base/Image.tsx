/** @jsx jsx */
import React from "react";
import { css, jsx } from "@emotion/core";
import ModalImage from "react-modal-image";
import classnames from "../../modules/classnames";
import { GREY_2 } from "../colors";

export const imageStyles = css`
  display: block;
  max-width: 100%;
  border: 1px solid ${GREY_2};
  margin: 29px auto;
  &.noborder {
    border: unset;
  }
`;

type Props = {
  src: string;
  alt: string;
  noborder?: boolean;
  expand?: boolean;
};

function Image({ src, alt, noborder, expand }: Props) {
  return (
    // using <React.Fragment> instead of the shorthand "<>...</>"
    // Storybook build was failing with error:
    // "transform-react-jsx: pragma has been set but pragmaFrag has not been set"
    <React.Fragment>
      {expand ? (
        <div css={imageStyles}>
          <ModalImage
            small={src}
            large={src}
            alt={alt}
            imageBackgroundColor="#fff"
            className="imageStyles"
          />
        </div>
      ) : (
        <img
          src={src}
          alt={alt}
          css={imageStyles}
          className={classnames({ noborder })}
        />
      )}
    </React.Fragment>
  );
}

export default Image;
