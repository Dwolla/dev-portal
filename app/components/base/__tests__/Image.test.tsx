import renderer from "react-test-renderer";
import Image from "../Image";

test("Image", () => {
  const tree = renderer
    .create(<Image src="image-src" alt="image-alt" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("Image no border", () => {
  const tree = renderer
    .create(<Image src="image-src" alt="image-alt" noborder />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("Image with expand", () => {
  const tree = renderer
    .create(<Image src="image-src" alt="image-alt" expand />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
