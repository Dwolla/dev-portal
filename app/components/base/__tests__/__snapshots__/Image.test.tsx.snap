// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image 1`] = `
<img
  alt="image-alt"
  className=" css-rv7724-imageStyles-imageStyles"
  src="image-src"
/>
`;

exports[`Image no border 1`] = `
<img
  alt="image-alt"
  className="noborder css-rv7724-imageStyles-imageStyles"
  src="image-src"
/>
`;

exports[`Image with expand 1`] = `
<div
  className="css-rv7724-imageStyles-imageStyles"
>
  <div>
    <img
      alt="image-alt"
      className="imageStyles"
      onClick={[Function]}
      src="image-src"
      style={
        {
          "cursor": "pointer",
          "maxHeight": "100%",
          "maxWidth": "100%",
        }
      }
    />
  </div>
</div>
`;
